src/components/application_manager/src/resumption/resumption_data_processor.cc

```diff
@@ -573,12 +573,76 @@ void ResumptionDataProcessor::AddCommands(
       application, application_manager_));
 }
 
+utils::Optional<ResumptionRequest> FindCommandRequest(
+    uint32_t command_id, std::vector<ResumptionRequest>& requests) {
+  using namespace utils;
+  auto request_it = std::find_if(
+      requests.begin(),
+      requests.end(),
+      [command_id](const ResumptionRequest& request) {
+        if (request.message[strings::msg_params].keyExists(strings::cmd_id) &&
+            (request.message[strings::msg_params].keyExists(
+                 strings::menu_params) ||
+             (request.message[strings::msg_params][strings::type] ==
+              hmi_apis::Common_VRCommandType::Command))) {
+          uint32_t cmd_id =
+              request.message[strings::msg_params][strings::cmd_id].asUInt();
+          return cmd_id == command_id;
+        }
+        return false;
+      });
+  if (requests.end() != request_it) {
+    return Optional<ResumptionRequest>(*request_it);
+  }
+  return Optional<ResumptionRequest>::OptionalEmpty::EMPTY;
+}
+
 void ResumptionDataProcessor::DeleteCommands(ApplicationSharedPtr application) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  const uint32_t app_id = application->app_id();
+  resumption_status_lock_.AcquireForReading();
+  std::vector<ResumptionRequest> failed_requests;
+  std::vector<ResumptionRequest> missed_requests;
+  if (resumption_status_.find(app_id) != resumption_status_.end()) {
```

Comment:
Extract it to function 

std::vector<ResumptionRequest> getFailedRequests(app_resumption_status)

Reply:
Done in e75db783045c452a32c33b86b646ff3cd2006d5b